packages/wrangler/src/__tests__/delete.test.ts

```diff
@@ -215,6 +227,156 @@ describe("delete", () => {
+				result: true,
+			});
+			writeWranglerToml();
+			mockListKVNamespacesRequest();
+			mockListReferencesRequest("test-name", {
+				services: {
+					incoming: [
+						{
+							service: "existing-worker",
+							environment: "production",
+							name: "BINDING",
+						},
+						{
+							service: "other-worker",
+							environment: "production",
+							name: "BINDING_TWO",
+						},
+					],
+					outgoing: [],
+				},
+				durable_objects: [
+					{
+						service: "do-binder",
+						environment: "production",
+						name: "ACTOR",
+						durable_object_namespace_id: "123",
+						durable_object_namespace_name: "actor_ns",
+					},
+					{
+						service: "test-name",
+						environment: "production",
+						name: "ACTOR",
+						durable_object_namespace_id: "123",
+						durable_object_namespace_name: "actor_ns",
+					},
+				],
+				dispatch_outbounds: [
+					{
+						service: "dispatcher",
+						environment: "production",
+						name: "DISPATCH",
+						namespace: "user-workers",
+						params: [],
+					},
+				],
+			});
+			mockListTailsByConsumerRequest("test-name", [
+				{
+					consumer: { script: "test-name" },
+					producer: { script: "i-make-logs" },
+					tag: "",
+					created_on: "",
+					modified_on: "",
+				},
+			]);
+			mockDeleteWorkerRequest({ force: true });
+			await runWrangler("delete");
+
+			expect(std).toMatchInlineSnapshot(`
+			      Object {
+			        "debug": "",
+			        "err": "",
+			        "info": "",
+			        "out": "Successfully deleted test-name",
+			        "warn": "",
+			      }
+		    `);
+		});
+
+		it("should not delete when extra confirmation to use force is denied", async () => {
+			mockConfirm({
+				text: `Are you sure you want to delete test-name? This action cannot be undone.`,
+				result: true,
+			});
+			mockConfirm({
+				text: `test-name is currently in use by other Workers:
+
+- Worker existing-worker (production) uses this Worker as a Service Binding
+
+You can still delete this Worker, but doing so WILL BREAK the Workers that depend on it. This will cause unexpected failures, and cannot be undone.
+Are you sure you want to continue?`,
+				result: false,
+			});
+			writeWranglerToml();
+			mockListKVNamespacesRequest();
+			mockListReferencesRequest("test-name", {
+				services: {
+					incoming: [
+						{
+							service: "existing-worker",
+							environment: "production",
+							name: "BINDING",
+						},
+					],
+					outgoing: [],
+				},
+			});
+			mockListTailsByConsumerRequest("test-name");
+			await runWrangler("delete");
+
+			expect(std).toMatchInlineSnapshot(`
+			      Object {
+			        "debug": "",
+			        "err": "",
+			        "info": "",
+			        "out": "",
+			        "warn": "",
+			      }
+		    `);
+		});
+
+		it("should not require confirmation when --force is used", async () => {
+			mockConfirm({
+				text: `Are you sure you want to delete test-name? This action cannot be undone.`,
+				result: true,
+			});
```

Comment:
If the user specifies `--force` I would expect that we should not even ask this question...